_projects/pmefr_project.md

```diff
@@ -0,0 +1,84 @@
+---
+layout: page_project
+title: Programming Model Extensions for Resilience
+date: 2016-02-10
+updated:
+navbar: Projects
+footer: true
+project_url:
+status: starting
+topics:
+  - resilience
+keywords:
+  -
+head: amer_a
+members:
+  - balaji_p
+  - beltran_v
+  - casas_m
+---
+
+## Research topic and goals
+
+The reliability of high performance computing systems is predicted to worsen given current
+technology trends, where several errors per day are expected to occur in exascale systems. 
+Consequently, the software stack that targets HPC systems needs to tolerate faults in order to cope with
+the instability of such systems. 
+Particularly, current parallel programming models, which assume that applications will execute on 
+failure-free systems, require resilience extensions.
+The most popular models used to program parallel systems are MPI for distributed memory
+and thread-based models, such as OpenMP, for shared-memory. 
+Several works have studied faulttolerance methods for MPI applications at the process level such as 
+the User-Level Fault Mitigation (ULFM) effort at ANL. 
+Recently, the need for finer-grained levels of resilience arose in the community
+because of the space and time overheads of coarse-grained approaches. 
+Consequently, research in this direction has become more active leading to new methods such as the rollbackrecovery
+method applied to asynchronous tasks in the OmpSs programming model.
+Despite recent advances, there exists no programming model that offers high degrees of resilience
+and low overheads for hybrid MPI+threads applications. 
+This collaboration aims at providing resilience extensions for hybrid MPI+threads programming models at multiple levels. 
+At the thread-level, errors are contained within the failing threads except if recovery is not possible,
+in which case propagating the failure to the MPI process will be necessary. 
+The system then will rely on process-level fault-tolerance techniques to mitigate the failure.
+
+## Results for 2014/2015
+
+There were interactions through telecon between the collaborators which involved discussions
+about ULFM for process-level resilience and the feasibility of thread-level fault-tolerance extensions.
+
+## Visits and meetings
+
+There were frequent telecon meetings since the beginning of the collaboration. There are no
+planned visits yet.
+
+## Impact and publications
+
+None yet.
+<!--
+
+-->
```

Comment:
Indeed, but you need to put a blank line before this, otherwise the html
page will actually display this!

On 10.02.16 16:52, Torbjörn Klatt wrote:
> In _projects/pmefr_project.md
> <https://github.com/JLESC/jlesc.github.io/pull/48#discussion_r52476168>:
> 
>> +## Results for 2014/2015
>> +
>> +There were interactions through telecon between the collaborators which involved discussions
>> +about ULFM for process-level resilience and the feasibility of thread-level fault-tolerance extensions.
>> +
>> +## Visits and meetings
>> +
>> +There were frequent telecon meetings since the beginning of the collaboration. There are no
>> +planned visits yet.
>> +
>> +## Impact and publications
>> +
>> +None yet.
>> +<!--
>> +
>> +-->
> 
> The HTML comment (|<!-- -->|) is there on purpose. Don't delete that.
> The JLESC references should go in there.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JLESC/jlesc.github.io/pull/48/files#r52476168>.
>